### PR TITLE
Addressing benchmark flakiness reporting

### PR DESCRIPTION
**Pull request by ChrisC**
After taking a closer look at the flakiness in some [previous benchmarks](https://github.com/GoogleChromeLabs/wpt-gen/pull/550), I noticed a couple of issues. We flag in our summary whenever more than 5 findings fell in the `mid (0.25, 0.75)` firing-rate band (aka "the flaky zone"). But the number of runs that could land in this range is highly dependent on the number of repeats:

- **3 repeats** → rates `0, 0.33, 0.67, 1.0`. "mid" = "fired once or twice of three" — maximally noisy; almost no chance to observe stability.
- **8 repeats** → "mid" = "fired 2–5 of 8" — a more genuinely-flaky signal observed over enough trials.

So if we ever wanted to gate on an absolute `>5` flakiness cutoff, we would be holding the smoke tier (less statistical power) to the same bar as a release run.

Additionally, investigating one corpus entry (`url-data-uri-fragment`) surfaced a deeper issue: flakiness was overwhelmingly caused by indeterminacy in assigning a particular rule to a finding, but not in the ability to detect the finding ("detection instability"). Defects in this test were found consistently _by line_ across repeats — the churn was in _which rule id_ got attached. We were counting our own taxonomy's ambiguity on top of the evaluator's ability to detect something wrong with the test.

### Proposed Changes to Consistency Scoring

**1. Decompose the signal (line vs rule_id).** Bucket findings by line, not by `(rule_id, line)`. This separates two things we were conflating:

- **Detection instability** — a line detected flakily across repeats.
- **Label churn** — a line detected consistently across repeats but labeled with ≥2 competing rules. I suggest we report this separately, but not score it.

**2. Switch to a graded stability score.** Replace the fixed-number flakiness count with a `0.0–1.0` detection-stability score (near-miss weighted, so a 4/8 finding counts as fully flaky and a 2/8 partially — dissolving the arbitrary bucket edges). The pass/warn/fail band **widens at low repeats** (`warn_at = 0.90 − 0.5/√reps`), so the smoke tier can't hard-fail flakiness on 3 samples. Grade is on detection instability only and the labeling churn stays advisory.

The consistency buckets section also now shows both a **graded band** (this run's target based on number of repeats) and a **fixed band** (comparable across runs; from the original report) so the two views are explicit.

### Addressing Instability and Churn Root Causes

This PR also includes a first pass at two high-impact rule/skill edits that cut the taxonomy overlap that inflated the signal:

- [Added a "no reaching" prohibition + fit-test to the evaluator skill](https://github.com/GoogleChromeLabs/wpt-gen/commit/bd4e931254b0189ec400b6f59cc1d250b0cf029f), curbing the model's tendency to stretch broad catch-all rules onto findings a more specific rule already owns.
- [Added a `notes` field to `rules.yaml`](https://github.com/GoogleChromeLabs/wpt-gen/commit/649a063e869335e466473bfc0a8624f0a0770655) and used it to disambiguate the CHECKLIST-004 / GENERAL-007 rules (invalid/no-op values → CHECKLIST-004; edge-case feature reliance → GENERAL-007).

In my test run, these two changes brought down the fixed flaky count on a smoke run from ~5 to 3 (sample report included below). I think further iteration could be made on the rules and eval skill to continue reducing model overreach and rule ambiguity. 

> Note: all of this affects **display/scoring grading** — `check_quality_gates` is unchanged. A future `--min-stability` gate could reuse this exact score.

### CODE REVIEW

File: scripts/benchmark/run_benchmark.py [hunk 1]
```diff
@@ -869,11 +1028,32 @@ def _entry_counts(entries: list[dict[str, Any]]) -> dict[EntryRole, int]:
...
 
+def _stability_bands(repeats: int) -> tuple[float, float]:
+    """(warn_at, fail_at) stability thresholds, widened at low repeats.
+
+    Base bands are warn <0.90 / fail <0.70; both shift down by ``0.5/√repeats``
+    so a low-repeat run needs a genuinely worse score to trip the same status
+    (you cannot confidently fail flakiness on few samples). At 8 repeats the
+    fail line is ~0.52; at 3 repeats ~0.41.
+    """
+    slack = 0.5 / (repeats**0.5) if repeats else 0.5
```
Comment (jcscottiii) [inline scripts/benchmark/run_benchmark.py]:
```suggestion
    slack = 0.5 / (repeats**0.5) if repeats > 0 else 0.5
```

Nit: repeats if repeats else 0.5 handles 0, but if a negative integer is ever passed, repeats**0.5 returns a complex number (j). repeats > 0 keeps this defensively guarded.

File: scripts/benchmark/analyze_consistency.py
Comment (jcscottiii) [inline scripts/benchmark/analyze_consistency.py:1]: We should try to add this to cloudbuild so that we get continued usage of this.

Comment (jcscottiii) [reply scripts/benchmark/analyze_consistency.py:1]: Also Gemini suggested something like this:

```suggestion
def _format_stability(val: Any) -> str:
    return f"{float(val):.2f}" if val is not None else "n/a"


def _run(out: Path) -> int:
    report_file = out / "report.json"
    if not report_file.is_file():
        sys.stderr.write(f"error: report file not found at {report_file}\n")
        return 1

    report = json.loads(report_file.read_text(encoding="utf-8"))

    for entry in report.get("entries", []):
        detection = entry.get("detection_flaky_lines", [])
        churn = entry.get("label_churn_lines", [])
        if not detection and not churn:
            continue
        d = entry.get("consistency_decomposition", {})
        stab = _format_stability(entry.get("stability"))
        print(f"\n{entry['entry_id']} ({entry.get('role', 'unknown')})")
        print(f"  stability {stab} | detection-flaky {d.get('line_flaky', 0)} | label-churn {d.get('label_churn', 0)}")
        for line in detection:
            print(f"    {'detection':<10} {line.get('line', 'file')}: {line.get('firings', 0)}/{line.get('repeats', 0)} ({_keys(line)})")
        for line in churn:
            print(f"    {'churn':<10} {line.get('line', 'file')}: {_keys(line)}")

    agg = report.get("aggregate", {})
    d = agg.get("consistency_decomposition", {})
    total_stab = _format_stability(agg.get("corpus_stability"))
    print(f"\nTOTALS  corpus-stability {total_stab} | detection-flaky {d.get('line_flaky', 0)} | label-churn {d.get('label_churn', 0)}")
    return 0
```

File: scripts/benchmark/run_benchmark.py [hunk 2]
```diff
@@ -467,6 +516,30 @@ def score_entry(
 ) -> EntryReport:
     """Scores a single entry from its loaded runs."""
     cons_rows = consistency_rows(runs)
+    line_rows = line_consistency_rows(runs)
+
+    def _line_label(lr: Any) -> str:
+        return f"L{lr.line_bucket[0]}" if lr.line_bucket else "file"
```
Comment (jcscottiii) [inline scripts/benchmark/run_benchmark.py]:
```suggestion
    def _line_label(lr: Any) -> str:
        if not lr.line_bucket:
            return "file"
        start, end = lr.line_bucket
        return f"L{start}" if start == end else f"L{start}-{end}"
```

Could we do something like this to get ranges?

**Review by jcscottiii — COMMENTED** [review]
Quick comments before my next meeting.

File: scripts/benchmark/analyze_consistency.py
Comment (ChrisC) [reply scripts/benchmark/analyze_consistency.py:1]: > We should try to add this to cloudbuild so that we get continued usage of this.

I went ahead and added an option `min-stability` gate in https://github.com/GoogleChromeLabs/wpt-gen/pull/551/commits/78d4fd61e5bf918636a54ba84037cacfeb871c1c !

**Review by jcscottiii — APPROVED** [review]
LGTM. Running cloud build workflow now
